x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/crawler/crawler_configuration/crawler_configuration.tsx

```diff
@@ -0,0 +1,104 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiSpacer,
+  EuiSplitPanel,
+  EuiSwitch,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { Status } from '../../../../../../../common/types/api';
+
+import { IndexViewLogic } from '../../index_view_logic';
+
+import { CrawlerConfigurationLogic } from './crawler_configuration_logic';
+
+export const CrawlerConfiguration: React.FC = () => {
+  const { htmlExtraction } = useValues(IndexViewLogic);
+  const { status } = useValues(CrawlerConfigurationLogic);
+  const { updateHtmlExtraction } = useActions(CrawlerConfigurationLogic);
+  return (
+    <>
+      <EuiSpacer />
+      <EuiSplitPanel.Outer hasBorder hasShadow={false}>
+        <EuiSplitPanel.Inner>
+          <EuiTitle size="s">
+            <h3>
+              {i18n.translate(
+                'xpack.enterpriseSearch.content.crawler.crawlerConfiguration.extractHTML.title',
+                { defaultMessage: 'Store full HTML' }
+              )}
+            </h3>
+          </EuiTitle>
+          <EuiSpacer />
+          <EuiText size="s">
+            <p>
+              {i18n.translate(
+                'xpack.enterpriseSearch.content.crawler.crawlerConfiguration.extractHTML.addExtraFieldDescription',
+                {
+                  defaultMessage:
+                    'Add an extra field in all documents with the value of the full HTML of the page being crawled.',
+                }
+              )}
+            </p>
+          </EuiText>
+          <EuiSpacer />
+          <EuiText size="s">
+            <p>
+              {i18n.translate(
+                'xpack.enterpriseSearch.content.crawler.crawlerConfiguration.extractHTML.increasedSizeWarning',
+                {
+                  defaultMessage:
+                    'This may dramatically increase the index size if the site being crawled is large.',
+                }
+              )}
+            </p>
+          </EuiText>
+        </EuiSplitPanel.Inner>
+        <EuiSplitPanel.Inner color="subdued">
+          <EuiFlexGroup>
+            <EuiFlexItem grow={false}>
+              <EuiSwitch
+                data-telemetry-id="entSearchContent-crawler-configuration-extractHtml"
+                label={i18n.translate(
+                  'xpack.enterpriseSearch.content.crawler.crawlerConfiguration.extractHTML.extractionSwitchLabel',
+                  {
+                    defaultMessage: 'Content extraction.',
+                  }
+                )}
+                disabled={status === Status.LOADING}
+                checked={htmlExtraction ?? false}
+                onChange={(event) => updateHtmlExtraction(event.target.checked)}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiLink>
```

Comment:
href coming soon? (mainly asking since we're so close to FF)

Maybe add a telemetry-id as well.

Reply:
Whoopsy, added that.

I'll add hrefs shortly after FF, that shouldn't be an issue. Have a bunch of them in the content plugin so I'm gonna hit them all in one go.